frontend/apps/reader/modules/readermenu.lua

```diff
@@ -218,11 +218,19 @@ dbg:guard(ReaderMenu, 'setUpdateItemTable',
     end)
 
 function ReaderMenu:exitOrRestart(callback)
-    self:onTapCloseMenu()
+    if self.menu_container then self:onTapCloseMenu() end
     UIManager:nextTick(function()
         self.ui:onClose()
         if callback ~= nil then
-            callback()
+            local waiting = function(waiting)
+                if self.ui and self.ui.document and self.ui.document:isEdited() then
+                    logger.dbg("waiting for save settings")
+                    UIManager:scheduleIn(1, function() waiting(waiting) end)
+                else
+                    callback()
+                end
+            end
+            UIManager:scheduleIn(1, function() waiting(waiting) end)
```

Comment:
Something in koreader is done in the background ? = not blocking main and single thread ? that you need to schedule and recheck ? (dunno much about this isEdited stuff, but that surprised me)

Reply:
It's not about isEdited, but otherwise it'll just go ahead and close all the widgets (including the ConfirmBox).

Or actually that's not quite right. It'll close the ReaderUI anyway. But either you close everything (including the ConfirmBox) or the program stays up until you dismiss some other widget on top. This seemed like the cleanest solution I could think of, but I'm open to suggestions. ;-)

Reply:
Ah, OK. So, you may wish to add there a comment, mentionning there may be a confirmbox that we wait for it to be dismissed (so future readers won't go on wondering is there's something happening in the background like i did :)